 AppOps: Avoid RuntimeException and log spam

We know Google Play Services have free reign and do whatever they want with the other Google apps.

This commit stops this logspam:

 AppOps  : Bad call: specified package com.google.android.gms under uid 10061 but it is really 10021
 AppOps  : Bad call: specified package android under uid 10036 but it is really 1000
 AppOps  : java.lang.RuntimeException: here
 AppOps  :      at com.android.server.AppOpsService.getOpsRawLocked(AppOpsService.java:1257)
 AppOps  :      at com.android.server.AppOpsService.checkPackage(AppOpsService.java:976)
 AppOps  :      at com.android.internal.app.IAppOpsService$Stub.onTransact(IAppOpsService.java:169)
 AppOps  :      at android.os.Binder.execTransact(Binder.java:565)

In my case, the 10061 app was Google Keep.

diff --git a/services/core/java/com/android/server/AppOpsService.java b/services/core/java/com/android/server/AppOpsService.java
index e31b4c6..70f9c00 100644
--- a/services/core/java/com/android/server/AppOpsService.java
+++ b/services/core/java/com/android/server/AppOpsService.java
@@ -1448,10 +1448,13 @@ private Ops getOpsRawLocked(int uid, String packageName, boolean edit) {
                     if (pkgUid != uid) {
                         // Oops!  The package name is not valid for the uid they are calling
                         // under.  Abort.
-                        RuntimeException ex = new RuntimeException("here");
-                        ex.fillInStackTrace();
-                        Slog.w(TAG, "Bad call: specified package " + packageName
-                                + " under uid " + uid + " but it is really " + pkgUid, ex);
+                        if (!"android".equals(packageName)) {
+                            // android is our overlord. Don't spam the log
+                            RuntimeException ex = new RuntimeException("Package uid doesn't match or it is not valid");
+                            ex.fillInStackTrace();
+                            Slog.w(TAG, "Bad call: specified package " + packageName
+                                    + " under uid " + uid + " but it is really " + pkgUid, ex);
+                        }
                         return null;
                     }
                 } finally {
